src/test/java/com/amazon/opendistroforelasticsearch/sql/intgtest/NestedFieldQueryTest.java

```diff
@@ -341,6 +355,37 @@ public void groupByNestedAndRegularField() {
         return kv("myNum", is(value));
     }
 
+    private Matcher<SearchHit> myNum(int[] values) {
+
+        return new BaseMatcher<SearchHit>() {
+
+            @Override
+            public boolean matches(Object item) {
+
+                if (item instanceof SearchHit) {
+                    final SearchHit hit = (SearchHit) item;
+                    List<Integer> actualValues = (ArrayList<Integer>) hit.getSourceAsMap().get("myNum");
+
+                    if (actualValues.size() != values.length) {
+                        return false;
+                    }
+                    for (int value : values) {
+                        if (!actualValues.contains(value)) {
```

Comment:
I don't think "contains" works here, it is not equal. For example, actualValues is [3, 3, 4] and values is [3, 4, 4]. The "if" condition will meet but they are different values.

Reply:
Agree, changing to 1-by-1 comparison of elements.